pkg/controller/common/certificates/ca_test.go

```diff
@@ -85,3 +90,36 @@ func TestNewSelfSignedCA(t *testing.T) {
 	require.Equal(t, testRSAPrivateKey, ca.PrivateKey)
 	require.True(t, ca.Cert.NotBefore.Before(time.Now().Add(2*time.Hour)))
 }
+
+func Test_HasPublicCA(t *testing.T) {
+	certsWithCA := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "c1-es-http-certs-public",
+		},
+		Data: map[string][]byte{
+			CAFileName: []byte("..."),
+			CertFileName: []byte("..."),
+		},
+	}
+	hasCA, err := HasPublicCA(k8s.NewFakeClient(&certsWithCA), esv1.ESNamer, "ns", "c1")
+	assert.NoError(t, err)
+	assert.True(t, hasCa)
+
+	certsWithoutCA := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "c2-es-http-certs-public",
+		},
+		Data: map[string][]byte{
+			CertFileName: []byte("..."),
+		},
+	}
+	hasCA, err = HasPublicCA(k8s.NewFakeClient(&certsWithoutCA), esv1.ESNamer, "ns", "c2")
+	assert.NoError(t, err)
+	assert.False(t, hasCA)
+
+	// no certs secret
+	_, err = HasPublicCA(k8s.NewFakeClient(), esv1.ESNamer, "ns", "c3")
+	assert.Error(t, err)
```

Comment:
Should we stick with the table tests for consistency?

Reply:
Yes.